ipa-core/src/protocol/ipa_prf/quicksort.rs

```diff
@@ -181,16 +179,16 @@ where
                     let rvl_ctx = rvl_ctx.clone();
                     let record_id = RecordId::from(idx);
                     async move {
-                        // Compare the current element against pivot and reveal the result.
-                        let comparison = compare_gt::<_, ThirtyTwoBitStep, SORT_CHUNK>(
+                        // Compare elements against pivot
+                        let comp = compare_gt::<_, ThirtyTwoBitStep, SORT_CHUNK>(
                             cmp_ctx, record_id, &k, &pivot,
                         )
-                        .await?
-                        .reveal(rvl_ctx, record_id) // reveal outcome of comparison
                         .await?;
+                        // Reveal the comparison result
+                        let revealed_comp = reveal(rvl_ctx, record_id, &comp).await?;
```

Comment:
There is no functional difference here, but calling the free function is necessary to work around rustc issue 100013.